src/client/QXmppPubSubManager.cpp

```diff
@@ -160,6 +163,50 @@ QXmppPubSubManager::~QXmppPubSubManager()
 {
 }
 
+///
+/// Requests all features of a pubsub service via service discovery.
+///
+/// This uses a \xep{0030, Service Discovery} info request to get a list of
+/// features.
+///
+/// \param serviceJid JID of the entity hosting the pubsub service
+/// \param serviceType type of service to retrieve features for
+/// \return
+///
+QFuture<QXmppPubSubManager::FeaturesResult> QXmppPubSubManager::requestFeatures(const QString &serviceJid, ServiceType serviceType)
+{
+    QXmppDiscoveryIq request;
+    request.setType(QXmppIq::Get);
+    request.setQueryType(QXmppDiscoveryIq::InfoQuery);
+    request.setTo(serviceJid);
+
+    return chainIq(client()->sendIq(std::move(request)), this, [=](QXmppDiscoveryIq &&iq) -> FeaturesResult {
+        const auto identities = iq.identities();
+
+        const auto isPubSubServiceFound = std::any_of(identities.cbegin(), identities.cend(), [=](const QXmppDiscoveryIq::Identity &identity) {
+            if (identity.category() == QStringLiteral("pubsub")) {
+                const auto identityType = identity.type();
+
+                switch (serviceType) {
+                case PubSubOrPep:
+                    return identityType == QStringLiteral("service") || identityType == QStringLiteral("pep");
+                case PubSub:
+                    return identityType == QStringLiteral("service");
+                case Pep:
+                    return identityType == QStringLiteral("pep");
+                }
+            }
+            return false;
+        });
+
+        if (isPubSubServiceFound) {
+            return iq.features().toVector();
```

Comment:
```suggestion
#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
            return iq.features();
#else
            return iq.features().toVector();
#endif
```
(toVector() has been removed in Qt 6)